include/inexor/vulkan-renderer/wrapper/device.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/availability_checks.hpp"
+#include "inexor/vulkan-renderer/settings_decision_maker.hpp"
+
+#include <spdlog/spdlog.h>
+#include <vulkan/vulkan.h>
+
+#include <cassert>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+/// @brief A RAII wrapper for VkDevice, VkPhysicalDevice and VkQueues.
+class Device {
+private:
+    VkDevice device = VK_NULL_HANDLE;
+    VkPhysicalDevice graphics_card = VK_NULL_HANDLE;
+    VkQueue graphics_queue = VK_NULL_HANDLE;
+    VkQueue present_queue = VK_NULL_HANDLE;
+    VkQueue transfer_queue = VK_NULL_HANDLE;
```

Comment:
Do we really need to initialize all of these to `VK_NULL_HANDLE`?

Reply:
Not really. If the destructor fails, an exception will be thrown anyways.
I will remove it.

Reply:
Done.